File: 2497-maximum-matching-of-players-with-trainers/maximum-matching-of-players-with-trainers.cpp
class Solution {
public:
    int matchPlayersAndTrainers(vector<int>& players, vector<int>& trainers) {
        sort(players.begin(), players.end(), greater<>());
        sort(trainers.begin(), trainers.end(), greater<>());
        int answer = 0;
        int trainers_idx = 0;
        int n = trainers.size();

        for(auto& player:players){
            if(trainers_idx < n && player <= trainers[trainers_idx]){
                answer++;
                trainers_idx++;
            }
        }
            
        return answer;
    }
};

File: LeetCode/2497-maximum-matching-of-players-with-trainers/maximum-matching-of-players-with-trainers.py
class Solution:
    def matchPlayersAndTrainers(self, players: List[int], trainers: List[int]) -> int:
        players.sort(reverse = True)
        trainers.sort(reverse = True)
        n = len(trainers)

        answer = 0
        trainers_idx = 0

        for player in players:
            if trainers_idx < n and player <= trainers[trainers_idx]:
                answer+=1
                trainers_idx+=1
        
        return answer
